### ISSUE TITLE
Operator "==" is (incorrectly) not supported on something with type of union of a `TypeVarTuple` 

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#5910]
**Describe the bug**
See code sample below.

I was expecting that `Union[*Ts]` should inherit all members from `object` so that "==" could work on it.

**Code or Screenshots**
```python
from typing import TypeVarTuple

Ts = TypeVarTuple("Ts")

def f(t0: tuple[*Ts], t1: tuple[*Ts], /):
    return all(v0 == v1 for v0, v1 in zip(t0, t1))  # Reported as not supported.
```

![image](https://github.com/microsoft/pyright/assets/15389744/9ed9187c-a053-4979-a3ce-4052f8077572)

**VS Code extension or command-line**
Pyright 1.1.326

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.327, which I just published. It will also be included in an upcoming release of pylance.
